traptor/traptor.py

```diff
@@ -854,18 +861,30 @@ def _enrich_tweet(self, tweet):
             # Store the limit count in Redis
             self._increment_limit_message_counter(limit_count=limit_count)
         elif self._message_is_tweet(tweet):
-            # Add the initial traptor fields
-            tweet = self._create_traptor_obj(tweet)
+            try:
+                # Add the initial traptor fields
+                tweet = self._create_traptor_obj(tweet)
+
+                # Add the created_at_iso field
+                tweet = self._add_iso_created_at(tweet)
+
+                # Add the rule information
+                enriched_data = self._find_rule_matches(tweet)
 
-            # Add the created_at_iso field
-            tweet = self._add_iso_created_at(tweet)
+                # Update the matched rule stats
+                if self.traptor_type != 'locations' and self.enable_stats_collection ==\
+                        'true':
+                    self._increment_rule_counter(enriched_data)
+            except Exception as e:
+                self.logger.error(traceback.format_exc())
+                self.logger.error("Failed to enrich tweet: {}".format(tweet))
```

Comment:
Put the `tweet` in extras, not the message otherwise the logging in kibana isn't useful. Also combine all three of these log statements into a single one:
```
self.logger.error("Failed to enrich tweet.",
                            {"tweet": json.dumps(tweet),
                             "ex": traceback.format_exc()})
```